### PR TITLE
Fix readme doc layout

### PR DESCRIPTION
**Pull request by feng-tao**
### Summary of Changes

The markdown in mkdocs has incorrect format (https://lyft.github.io/amundsen/) on certain sections. Fix them for now.

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely.
- [ ] PR includes a summary of changes.
- [ ] My commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"

### CODE REVIEW

**Comment by Golodhros** [comment]
@all-contributors please add @feng-tao for bug, code, data, doc, example, maintenance, review, test

File: README.md [hunk 1]
```diff
@@ -72,19 +72,35 @@ Please visit the Amundsen installation documentation for a [quick start](https:/
...
 - Dashboards
 
-Amundsen can connect to any database that provides `dbapi` or `sql_alchemy` interface (which most DBs provide).
-The databases we see most frequently used in the community are:
-- Hive and anything that works with Hive metastore (Spark SQL, Presto, Athena, etc.)
-- BigQuery
-- PostgreSQL and anything that uses the same interface (Redshift)
-- Snowflake
+## Supported Connectors
+
```
Comment (Golodhros) [inline README.md:81]: IMHO this is a great place to mention what a "connector" is. Just in one line if possible, something like:
``` Connectors are .... . We support many connectors for either tables or dashboards, following there is a list of them.```

Comment (feng-tao) [reply README.md:81]: yeah, I found the issue that somehow the markdown is broken in mkdocs(https://lyft.github.io/amundsen/ if it is not followed by bullet point but a list sentences... I haven't figured out why that's the case.

Comment (jornh) [reply README.md:81]: @feng-tao it’s probably due to that the GitHub Markdown renderer is more forgiving than the Python based one in mkdocs. 

Perhaps some Markdown listing is needed like https://dev.to/jonasbn/blog-post-getting-to-know-github-actions-3m61 which has pointers/info to both a travis based setup and GitHub actions.

Comment (feng-tao) [reply README.md:81]: thanks @jornh for the info!

**Pull request merged**